akka-cluster-sharding-typed/src/test/scala/docs/delivery/ShardingDocExample.scala

```diff
@@ -0,0 +1,217 @@
+    import TodoService._
+
+    private implicit val askTimeout: Timeout = 5.seconds
+
+    private def start(
+        producerController: ActorRef[ShardingProducerController.Start[TodoList.Command]]): Behavior[Command] = {
+      val requestNextAdapter: ActorRef[ShardingProducerController.RequestNext[TodoList.Command]] =
+        context.messageAdapter(WrappedRequestNext.apply)
+      producerController ! ShardingProducerController.Start(requestNextAdapter)
+
+      Behaviors.receiveMessagePartial {
+        case WrappedRequestNext(next) =>
+          active(next)
+        case UpdateTodo(_, _, _, replyTo) =>
+          // not hooked up with shardingProducerController yet
+          replyTo ! Rejected
+          Behaviors.same
+      }
+    }
+
+    private def active(requestNext: ShardingProducerController.RequestNext[TodoList.Command]): Behavior[Command] = {
+      Behaviors.receiveMessage {
+        case WrappedRequestNext(next) =>
+          active(next)
+
+        case UpdateTodo(listId, item, completed, replyTo) =>
+          if (requestNext.bufferedForEntitiesWithoutDemand.getOrElse(listId, 0) >= 100)
+            replyTo ! Rejected
+          else {
+            val requestMsg = if (completed) TodoList.CompleteTask(item) else TodoList.AddTask(item)
+            context.ask[ShardingProducerController.MessageWithConfirmation[TodoList.Command], Done](
+              requestNext.askNextTo,
+              askReplyTo => ShardingProducerController.MessageWithConfirmation(listId, requestMsg, askReplyTo)) {
+              case Success(Done) => Confirmed(replyTo)
+              case Failure(_)    => TimedOut(replyTo)
+            }
+          }
+          Behaviors.same
+
+        case Confirmed(originalReplyTo) =>
+          originalReplyTo ! Accepted
+          Behaviors.same
+
+        case TimedOut(originalReplyTo) =>
+          originalReplyTo ! MaybeAccepted
+          Behaviors.same
+      }
+    }
+
+  }
+  //#producer
+
+  def illustrateInit(): Unit = {
+    Behaviors.setup[Nothing] { context =>
+      //#init
+      import akka.cluster.sharding.typed.scaladsl.ClusterSharding
+      import akka.cluster.sharding.typed.scaladsl.Entity
+      import akka.cluster.sharding.typed.scaladsl.EntityTypeKey
+      import akka.cluster.typed.Cluster
+
+      val db: DB = ???
+
+      val system = context.system
+
+      val TypeKey = EntityTypeKey[ConsumerController.SequencedMessage[TodoList.Command]]("todo")
+
+      val region = ClusterSharding(system).init(Entity(TypeKey)(entityContext =>
+        ShardingConsumerController(start => TodoList(entityContext.entityId, db, start))))
+
+      val selfAddress = Cluster(system).selfMember.address
+      val producerId = s"todo-producer-${selfAddress.host}:${selfAddress.port}"
+
+      val producerController =
+        context.spawn(ShardingProducerController(producerId, region, durableQueueBehavior = None), "producerController")
+
+      context.spawn(TodoService(producerController), "producer")
```

Comment:
Wondering if you could also include an example of actually calling UpdateTodo in the TodoService? What would that look like?

Reply:
Try "todoService" in the EntityTypeKey instead, because I think the example is already using "todo" for the `TodoList` and here you are trying to interact with the `TodoService`.

However, I think you misunderstand how it is intended to be used. You try to use it with two levels of Sharding, one for the TodoList and one for the TodoService. The docs say " A single ShardingProducerController per ActorSystem (node) can be shared for sending to all entities of a certain entity type." https://doc.akka.io/docs/akka/current/typed/reliable-delivery.html#sharding

If you have more questions I'd recommend switching over to https://discuss.akka.io/